### DISCUSSION TITLE
runReactants with both reactants matched and no warnings on the reaction templates produces 0 molecules

### DISCUSSION THREAD

**Discussion by wirmius** [rdkit/rdkit#4090] [Q&A]
I have had a problem with my code that I managed to boil down to this:

```import rdkit
import rdkit.Chem as Chem
import rdkit.Chem.AllChem as AllChem
from rdkit.Chem import rdChemReactions 
from rdkit.Chem.rdChemReactions import ChemicalReaction

rxn = AllChem.ReactionFromSmarts('Cl-[C&H0&D3&+0:1](=[O&D1&H0:2])-[c:3].[C:4]-[O&H1&D1&+0:5]>>[C:4]-[O&H0&D2&+0:5]-[C&H0&D3&+0:1](=[O&D1&H0:2])-[c:3]')
rdChemReactions.SanitizeRxn(rxn, Chem.SANITIZE_ALL)
rxn.Initialize()

t_0, t_1 = list(rxn.GetReactants())
m = Chem.MolFromSmiles('CCO')
m2 = Chem.MolFromSmiles('c1ccccc1C(=O)Cl')

m.HasSubstructMatch(t_0), m.HasSubstructMatch(t_1)
# (False, True)
m2.HasSubstructMatch(t_0), m2.HasSubstructMatch(t_1)
# (True, False)
p = rxn.RunReactants((m, m2))
# ()  <-- 0 products
```
Since the substructures for both of the reagents had been matched, I'd expect it to have something in the products. Why doesnt it? Am I missing a step of reaction preprocessing?

**Configuration (please complete the following information):**
 - RDKit version: 2020.09.5
 - OS: Ubuntu 20.04
 - Python version (if relevant): 3.7
 - Are you using conda? Yes
 - If you are using conda, which channel did you install the rdkit from? conda-forge

**Additional context**
The reaction template is from a set of templates that are known to be functional. This happens sometimes, for some reactions in the set that I have.

PS: I know its probably some stupid preprocessing step that I have missed, I would be very grateful if you had a look at it.
PPS: I attached a pdf of a notebook that demonstrates my problem (with different template this time).
[template_fix.pdf](https://github.com/rdkit/rdkit/files/6433298/template_fix.pdf)

**Comment by greglandrum** [comment]
@wirmius : `rxn.RunReactants()` expects you to provide the reactant molecules in the same order as the reactant templates, so you need to reverse the order of the arguments:
```
In [4]: rxn.RunReactants((m, m2))
Out[4]: ()

In [5]: rxn.RunReactants((m2, m))
Out[5]: ((<rdkit.Chem.rdchem.Mol at 0x7fb23c666c80>,),)
```
